test/end-to-end/offer.js

```diff
@@ -479,6 +479,42 @@ describe("Offer endpoint tests", () => {
                 expect(created_offer).toHaveProperty("publishDate");
             });
         });
+
+        describe("Job Duration", () => {
+            test("should fail if jobMinDuration is greater than jobMaxDuration", async () => {
+                const offer_params = generateTestOffer({
+                    jobMinDuration: 10,
+                    jobMaxDuration: 8,
+                    owner: test_company._id,
+                    ownerName: test_company.name,
+                });
+
+                const res = await request()
+                    .post("/offers/new")
+                    .send(withGodToken(offer_params));
+
+                expect(res.status).toBe(HTTPStatus.UNPROCESSABLE_ENTITY);
+                expect(res.body).toHaveProperty("error_code", ErrorTypes.VALIDATION_ERROR);
+                expect(res.body).toHaveProperty("errors");
+                expect(res.body.errors[0]).toHaveProperty("param", "jobMaxDuration");
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.MUST_BE_AFTER("jobMinDuration"));
+            });
+
+            test("should succeed if jobMaxDuration is greater than jobMinDuration", async () => {
+                const offer_params = generateTestOffer({
+                    jobMinDuration: 8,
+                    jobMaxDuration: 10,
+                    owner: test_company._id,
+                    ownerName: test_company.name,
+                });
+
+                const res = await request()
+                    .post("/offers/new")
+                    .send(withGodToken(offer_params));
+
+                expect(res.status).toBe(HTTPStatus.OK);
+            });
+        });
```

Comment:
add a test for when jobMinDuration is not provided (it should pass)

Reply:
I want to check something out.
```
jobMinDuration: {
    type: Number,
    required: function() {
        // jobMinDuration is required if jobMaxDuration was specified
        return !!this.jobMaxDuration;
    },
},
```
As you can see, in the model validator there's a comment (and function) saying that the jobMinDuration is required in case jobMaxDuration was specified, which goes against your current request.
This way, I think the scenario would be (based on the current model):

- jobMinDuration and jobMaxDuration both specified  => pass
- jobMinDuration unspecified and jobMaxDuration specified => fail
- jobMinDuration specified and jobMaxDuration unspecified => pass
- jobMinDuration and jobMaxDuation both unspecified => pass

Is this intended or do you want me to change something?

Reply:
Yep, sorry, I was thinking about the edit scenario, where jobMinDuration can be omitted and it will use the one currently in the offer.

Your bulletpoints are correct.